### PR TITLE
Allow freezing a texture.

### PR DESCRIPTION
**Pull request by amirh**
This is needed to avoid jank when resizing an embedded Android view.
See https://github.com/flutter/flutter/issues/19572#issuecomment-410400724 for more details.

### CODE REVIEW

**Comment by amirh** [comment]
@chinmaygarde I'm not sure what would be the iOS equivalent of skipping `SurfaceTextueUpdateTexImage`, any ideas?
(I only need this on Android, but it will be nice to keep it symmetric on iOS as well)

**Comment by amirh** [comment]
Implemented the freeze for iOS as well, @chinmaygarde want to take a look?

File: lib/ui/compositing.dart [hunk 1]
```diff
@@ -235,11 +235,13 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   /// Adds a backend texture to the scene.
   ///
   /// The texture is scaled to the given size and rasterized at the given offset.
-  void addTexture(int textureId, { Offset offset: Offset.zero, double width: 0.0, double height: 0.0 }) {
+  ///
+  /// If `freeze` is true new frames will not be consumed from the texture.
```
Comment (chinmaygarde) [inline lib/ui/compositing.dart]: Can this docstring be more detailed please?

Comment (amirh) [reply lib/ui/compositing.dart]: Sure, can you help me by explaining what wasn't clear when you read this?

Comment (chinmaygarde) [reply lib/ui/compositing.dart]: * Who not consume the textures?
* When should the flag be set?

Comment (amirh) [reply lib/ui/compositing.dart]: Thanks, rewrote the comment PTAL.

**Comment by amirh** [comment]
Woops had a bad merge fixing

**Pull request merged**